serverless/serverless
# Option to disable checking of API Gateway CloudWatch role setting

## What did you implement

Provides a `roleManagedExternally` setting to enable/disable checking of API Gateway CloudWatch role when using `provider.logs` to enable API Gateway logs. This avoids creating a custom resource lambda (which requires a number of extra permissions assigned to your cfnRole just to run it). You must ensure the setting is there through other means, otherwise the deployment will fail when trying to enable the logs.

Also documents the logging choices available under the Logs section in the API Gateway events doc page.

See also discussion [here](https://github.com/serverless/serverless/issues/6973#issuecomment-584828107).

Closes #6973

## How can we verify it

```yml
provider:
  logs:
    restApi:
       roleManagedExternally: true
```

Deploying a serverless.yml with roleManagedExternally: false should **not** create a custom resource. Deployment should succeed when API Gateway already has a CloudWatch role ARN configured in that region (and fail otherwise, obviously).

Default is false, to avoid making it a breaking change.

## Todos

<details>
<summary>Useful Scripts</summary>


- `npm run test:ci` --> Run all validation checks on proposed changes
- `npm run lint:updated` --> Lint all the updated files
- `npm run lint:fix` --> Automatically fix lint problems (if possible)
- `npm run prettier-check:updated` --> Check if updated files adhere to Prettier config
- `npm run prettify:updated` --> Prettify all the updated files

</details>


**_Is this ready for review?:_** YES
**_Is it a breaking change?:_** NO
